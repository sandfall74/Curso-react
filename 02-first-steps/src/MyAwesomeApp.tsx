import type { CSSProperties } from "react";

const firstName = 'Adrian';
const lastName = 'Piedra'
const favoriteGame =['Elden Ring', 'Smash','Metal Gear'];
const isActive = true;
const address={
    zipCode:'ABC-123',
    country:'Canada'
}
const myStyle: CSSProperties={
    backgroundColor:'#fafafa',
        borderRadius: 10,
        padding: 10,
}
export const MyAwesomeApp=()=>{

return (
<div data-testid="div-app">

<h1 data-testid="first-name-tittle">{firstName} </h1>
<h3>{lastName}</h3>
<p>{favoriteGame.join(', ')}</p>
<h1>{isActive ? 'Activo': 'No Activo'}</h1>
<p
    style={myStyle}
>{JSON.stringify(address)}</p>

</div>
)
}