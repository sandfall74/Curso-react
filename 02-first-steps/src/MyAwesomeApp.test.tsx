import { describe, test,expect} from "vitest";
import {render,screen} from '@testing-library/react';;
import { MyAwesomeApp } from "./MyAwesomeApp";

describe('MyAwesomeApp',()=>{
    test('should render first name and last name',()=>{
        const {container}=render( <MyAwesomeApp/>);

        //screen.debug();

        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');
        console.log(h1?.innerHTML);
        expect(h1?.innerHTML).toContain('Adrian');
         expect(h3?.innerHTML).toContain('Piedra');
    });
    test('should render first name and last name-screen',()=>{
       render( <MyAwesomeApp/>);

        screen.debug();

       /*  const h1 = screen.getByRole('heading',{
            level:1
        }); */

        const h1 = screen.getByTestId('first-name-tittle');
        expect(h1?.innerHTML).toContain('Adrian');
    });
    test('should match snpachot',()=>{
        const {container}=render( <MyAwesomeApp/>);
        expect(container).toMatchSnapshot();
    })
    test('should match snpachot2',()=>{
        render( <MyAwesomeApp/>);
        expect(screen.getByTestId('div-app')).toMatchSnapshot();
    })
})