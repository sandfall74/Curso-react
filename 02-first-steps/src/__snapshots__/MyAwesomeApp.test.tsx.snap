// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snpachot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-tittle"
    >
      Adrian
       
    </h1>
    <h3>
      Piedra
    </h3>
    <p>
      Elden Ring, Smash, Metal Gear
    </p>
    <h1>
      Activo
    </h1>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 10px; padding: 10px;"
    >
      {"zipCode":"ABC-123","country":"Canada"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snpachot2 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-tittle"
  >
    Adrian
     
  </h1>
  <h3>
    Piedra
  </h3>
  <p>
    Elden Ring, Smash, Metal Gear
  </p>
  <h1>
    Activo
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 10px; padding: 10px;"
  >
    {"zipCode":"ABC-123","country":"Canada"}
  </p>
</div>
`;
